lib/gringotts/gateways/mercadopago.ex

```diff
@@ -0,0 +1,376 @@
+  def void(payment_id, opts) do
+    # commit(args, ...)
+  end
+
+  @doc """
+  Refunds the `amount` to the customer's account with reference to a prior transfer.
+
+  > Refunds are allowed on which kinds of "prior" transactions?
+
+  ## Note
+
+  > The end customer will usually see two bookings/records on his statement. Is
+  > that true for mercadopago?
+  > Is there a limited time window within which a void can be perfomed?
+
+  ## Example
+
+  > A barebones example using the bindings you've suggested in the `moduledoc`.
+  """
+  @spec refund(Money.t, String.t(), keyword) :: {:ok | :error, Response}
+  def refund(amount, payment_id, opts) do
+    # commit(args, ...)
+  end
+
+  @doc """
+  Stores the payment-source data for later use.
+
+  > This usually enable "One Click" and/or "Recurring Payments"
+
+  ## Note
+
+  > If there's anything noteworthy about this operation, it comes here.
+
+  ## Example
+
+  > A barebones example using the bindings you've suggested in the `moduledoc`.
+  """
+  @spec store(CreditCard.t(), keyword) :: {:ok | :error, Response}
+  def store(%CreditCard{} = card, opts) do
+    # commit(args, ...)
+  end
+
+  @doc """
+  Removes card or payment info that was previously `store/2`d
+
+  Deletes previously stored payment-source data.
+
+  ## Note
+
+  > If there's anything noteworthy about this operation, it comes here.
+
+  ## Example
+
+  > A barebones example using the bindings you've suggested in the `moduledoc`.
+  """
+  @spec unstore(String.t(), keyword) :: {:ok | :error, Response}
+  def unstore(registration_id, opts) do
+    # commit(args, ...)
+  end
+
+  ###############################################################################
+  #                                PRIVATE METHODS                              #
+  ###############################################################################
+  
+  # Makes the request to mercadopago's network.
+  # For consistency with other gateway implementations, make your (final)
+  # network request in here, and parse it using another private method called
+  # `respond`.
+  @spec commit(_) :: {:ok | :error, Response}
+  defp commit(_) do
+    # resp = HTTPoison.request(args, ...)
+    # respond(resp, ...)
+  end
+
+  # Parses mercadopago's response and returns a `Gringotts.Response` struct
+  # in a `:ok`, `:error` tuple.
+  @spec respond(term) :: {:ok | :error, Response}
+  defp respond(mercadopago_response)
+  defp respond({:ok, %{status_code: 200, body: body}}), do: "something"
+  defp respond({:ok, %{status_code: status_code, body: body}}), do: "something"
+  defp respond({:error, %HTTPoison.Error{} = error}), do: "something"
+
+  defp get_customer_id(opts) do
+    body = %{"email": opts[:email]} |> Poison.encode!
+    headers = [{"content-type", "application/json"}, {"accept", "application/json"}]
+    response = HTTPoison.post!("#{@base_url}/v1/customers?access_token=#{opts[:config][:access_token]}", body, headers, [])
```

Comment:
- I think you should use 

> `HTTPoison.post`

 rather than 

> `HTTPoison.post!`

 as the latter will raise an exception in case it fails.
- Also I would suggest you to handle all request via the `commit` function